test/factories/file_submissions.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :file_submission do
+    challenge_id { "" }
+    user_id { "" }
```

Comment:
This is not the correct way to setup the `challenge_id` and `user_id` on the Factory, you can just use the shorthand and just say `user` instead of `user_id { "" }`. For the `challenge_id {}`, that should always be a file_submission_challenge, so you should be able to say `challenge { create(:file_submission_challenge) }`.